File: samples/dotnet/eftl/Request.cs
/*
 * Copyright (c) 2013-$Date: 2015-07-31 16:05:02 -0700 (Fri, 31 Jul 2015) $ TIBCO Software Inc.
 * All Rights Reserved.
 * For more information, please contact:
 * TIBCO Software Inc., Palo Alto, California, USA
 */

using System;
using System.Threading;
using System.Collections;

using TIBCO.EFTL;

public class Request
{
    static String url = "ws://localhost:8585/channel";
    static String username = "user";
    static String password = "password";

    public class RequestListener: IRequestListener
    {
        IConnection connection;

        public RequestListener(IConnection connection)
        {
            this.connection = connection;
        }

        public void OnReply(IMessage reply) 
        {
            Console.WriteLine("Reply message " + reply);

            // Disconnect from the server.
            connection.Disconnect();
        }
    
        public void OnError(IMessage message, int code, String reason) 
        {
            Console.WriteLine("Request error: " + reason);

            // Disconnect from the server.
            connection.Disconnect();
        }
    } 

    public class ConnectionListener: IConnectionListener
    {
        public void OnConnect(IConnection connection)
        {
            Console.WriteLine("Connected to server at " + url);

            try
            {
                // Create request message.
                IMessage request = connection.CreateMessage();
                request.SetString("type", "request");
                request.SetString("text", "This is a sample request message");

                // Asynchronously publish the request with a 10 second timeout.
                connection.SendRequest(request, 10.0, new RequestListener(connection));
            } 
            catch (Exception e)
            {
                Console.WriteLine(e.ToString());
            }
        }

        public void OnDisconnect(IConnection connection, int code, String reason)
        {
            Console.WriteLine("Disconnected from server: " + reason);
        }

        public void OnError(IConnection connection, int code, String reason)
        {
            Console.WriteLine("Connection error " + reason);
        }

        public void OnReconnect(IConnection connection)
        {
            Console.WriteLine("Reconnected to server at " + url);
        }
    } 

    public static void Main (string []args) 
    {
        if (args.Length > 0)
        {
            url = args[0];
        }

        Console.Write("#\n# Request\n#\n# {0}\n#\n", EFTL.GetVersion());

        Hashtable options = new Hashtable();

        if (username != null)
            options.Add(EFTL.PROPERTY_USERNAME, username);

        if (password != null)
            options.Add(EFTL.PROPERTY_PASSWORD, password);

        try
        {
            // Asynchronously connect to the server.
            EFTL.Connect(url, options, new ConnectionListener());
        } 
        catch (Exception e)
        {
            Console.WriteLine(e.ToString());
        }
    }
}

File: samples/nodejs/eftl/request.js
/*
 * Copyright (c) 2013-$Date: 2019-06-21 12:47:57 -0700 (Fri, 21 Jun 2019) $ TIBCO Software Inc.
 * Licensed under a BSD-style license. Refer to [LICENSE]
 * For more information, please contact:
 * TIBCO Software Inc., Palo Alto, California, USA
 */

'use strict'

const eFTL = require('eftl');

const url = (process.argv[2] || "ws://localhost:8585/channel");

console.log(eFTL.getVersion());

// Start a connection to the server.
//
// The username and password are required when connecting
// to a server that has been configured for JAAS.
//
// The onConnect function is invoked following a successful
// connection to the server. 
//
// The onDisconnect function is invoked following a failed
// connection attempt to the server.
//
eFTL.connect(url, {
    username: 'user',
    password: 'password',
    trustAll: true,
    onConnect: connection => {

        console.log("Connected");

        try {
            // create a request message
            var msg = connection.createMessage();
            msg.set('type', 'request');
            msg.set('text', 'This is a sample eFTL request message');

            // send a request message and wait 10 seconds for a reply
            connection.sendRequest(msg, 10000, {

                onReply: reply => {

                    console.log('Received reply: ' + reply.toString());

                    connection.disconnect();
                },

                onError: (request, code, reason) => {

                    console.log('Request error: ' + reason);

                    connection.disconnect();
                }
            });
        } catch (err) {
            console.log(err);

            connection.disconnect();
        }
    },
    onDisconnect: (connection, code, reason) => {

        console.log(reason);
    },
    onError: (connection, code, reason) => {

        console.log(reason);
    }
});
